### PR TITLE
Add akka-http-oauth2-provider sub project.

### PR DESCRIPTION
**Pull request by cdiniz**
Hi guys. I have been working on lightbend templates (play+slick+oauth2,akka-http-slick-oauth2) using your great scala-oauth2-provider. Since you don't have any helper for akka http, I decided to do it. What do you think? I have a sample template using the same helper:  https://github.com/cdiniz/slick-akka-http-oauth2.

P.S. I'll work on tests for the accessTokenRoute asap.

### CODE REVIEW

**Comment by tsuyoshizawa** [comment]
Great! I'm not a akka-http user so this PR would help for akka-http users.

> I'll work on tests for the accessTokenRoute asap.

I see. Let me know when you complete the work.

File: akka-http-oauth2-provider/src/main/scala/scalaoauth2/provider/Oauth2Provider.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import scala.util.{ Failure, Success }
+
+trait OAuth2Provider[U] extends Directives with DefaultJsonProtocol {
+  import OAuth2Provider.tokenResponseFormat
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+
+  val oauth2DataHandler: DataHandler[U]
+
+  val tokenEndpoint = new TokenEndpoint {
+    override val handlers = Map(
```
Comment (tsuyoshizawa) [inline akka-http-oauth2-provider/src/main/scala/scalaoauth2/provider/Oauth2Provider.scala]: All authentications are enabled by default, it is not good to keep security.
Developers will use authentication they do not use without being aware.

Could you change this like [Play subproject](https://github.com/nulab/scala-oauth2-provider/blob/0.17.2/play2-oauth2-provider/src/main/scala/scalaoauth2/provider/OAuth2Provider.scala#L16)?

**Comment by cdiniz** [comment]
Hello. I removed the default tokenEndpoint as you suggested, and I added tests for the helper route. Is it ok for you?

**Comment by tsuyoshizawa** [comment]
> Is it ok for you?

@cdiniz Yes. Thank you.

Could you rename `Oauth2Provider.scala` to `OAuth2Provider.scala`?
Class name is `OAuth2Provider` but file name is different.

**Comment by cdiniz** [comment]
done!

**Pull request merged**